tests/CMakeLists.txt

```diff
@@ -82,6 +82,15 @@ foreach(test IN LISTS testFileSuffixes)
                       LIBS      metkit)
 endforeach()
 
+# Compile C test
+ecbuild_add_test( TARGET        metkit_test_c_compiled
+                  SOURCES       test_c_api.c
+                  INCLUDES      "${ECKIT_INCLUDE_DIRS}"
+                  LIBS          metkit
+                  NO_AS_NEEDED
+                  ENVIRONMENT   "${metkit_env}"
+                  LINKER_LANGUAGE C)
+
```

Comment:
It is not required to set `LINKER_LANGUAGE C` as this will be detected based on the file extension. If you still want to be explicit its probably better to set the language for the source file to 'C' as we want the source file to pass trough the C-Compilation, i.e. `set_source_files_properties(test_c_api.c PROPERTIES LANGUAGE C)`

You should also probably also add C to the project languages. Right now C is always enabled due to how we pull in ecbuild and hence it works but it is surprising to see that we invoke the C-Compiler although the make project explicitly only requests C++.
I.e. the top-level project declaration should become `project( metkit LANGUAGES CXX )`